temporal-sdk/src/main/java/io/temporal/client/schedules/Schedule.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client.schedules;
+
+import java.util.Objects;
+import javax.annotation.Nonnull;
+
+/** A schedule for periodically running an action. */
+public final class Schedule {
+
+  public static Schedule.Builder newBuilder(
```

Comment:
This is intentional so a user does not create a schedule without an `action` or `spec`

Reply:
Right, but what if I have an action, a spec, a state, and a policy. Can't I call an overload with all 4 instead of just these two? Maybe there doesn't need to be a builder for schedules that are just these 4 values.

Reply:
I didn't add because it would require more overloads (3 more or 1 and allow nulls) , If we want to add it that's fine but I'd remove the builder in that case.

Reply:
I think that it's reasonable to add the overloads or put a builder on both. Whether I choose to use calendar spec or interval spec, the timing spec objects should be able to be created the same way. They are just objects with fields after all, they are not that different.

Reply:
I think you meant to comment this on a different thread?

Reply:
Actually I remembered I already tried to not use a builder but it made updating the schedule and have constant objects difficult. It also is just not consistent with how the rest of the Java SDK is written even though I don't like it